tests/functional/BaseCase.php

```diff
@@ -361,11 +362,46 @@ private function dropTestBucket(
         $nameGenerator = new NameGenerator($this->getStackPrefix());
         $datasetName = $nameGenerator->createObjectNameForBucketInProject($bucketId, $branchId);
         $bqClient = $this->clientManager->getBigQueryClient($this->testRunId, $projectCredentials);
-        $dataset = $bqClient->dataset($datasetName);
-        try {
-            $dataset->delete(['deleteContents' => true]);
-        } catch (Throwable) {
-            // ignore
+
+        if (!$bqClient->dataset($datasetName)->exists()) {
+            return;
+        }
+
+        // Try deletion with retry
+        $maxRetries = 3;
+        $lastException = null;
+        for ($i = 0; $i < $maxRetries; $i++) {
+            try {
+                $bqClient->dataset($datasetName)->delete(['deleteContents' => true]);
+                // Verify deletion - need fresh reference to check actual state
+                // @phpstan-ignore booleanNot.alwaysFalse (exists() result changes after delete)
+                if (!$bqClient->dataset($datasetName)->exists()) {
+                    return;
+                }
+                // Dataset still exists, wait and retry
+                usleep(500000); // 0.5 second
+            } catch (Throwable $e) {
+                $lastException = $e;
+                $this->log->add(sprintf(
+                    'dropTestBucket attempt %d failed for %s: %s',
+                    $i + 1,
+                    $datasetName,
+                    $e->getMessage(),
+                ));
+                usleep(500000); // 0.5 second before retry
+            }
+        }
```

Comment:
Na retry máme našu retry lib je tu různě použitá třeba v https://github.com/keboola/php-storage-driver-bigquery/blob/main/src/Handler/Workspace/Create/CreateWorkspaceHandler.php#L97-L119 téj bysme sa měli držat než tady těch custom vibe coded retry